superset/tasks/schedules.py

```diff
@@ -409,6 +413,61 @@ def schedule_email_report(  # pylint: disable=unused-argument
     else:
         raise RuntimeError("Unknown report type")
 
+@celery_app.task(
+    name="alerts.run_query",
+    bind=True,
+    soft_time_limit=config["EMAIL_ASYNC_TIME_LIMIT_SEC"],
+)
+def schedule_alert_query(  # pylint: disable=unused-argument
+    task: Task,
+    report_type: ScheduleType,
+    schedule_id: int,
+    recipients: Optional[str] = None,
+) -> None:
+    model_cls = get_scheduler_model(report_type)
+    dbsession = db.create_scoped_session()
+    schedule = dbsession.query(model_cls).get(schedule_id)
+
+    # The user may have disabled the schedule. If so, ignore this
+    if not schedule or not schedule.active:
+        logger.info("Ignoring deactivated alert")
+        return
+
+    if report_type == ScheduleType.alert:
+        if run_alert_query(schedule, dbsession):
+            # deliver_dashboard OR deliver_slice
+            return
+    else:
+        raise RuntimeError("Unknown report type")
+
+
+def run_alert_query(alert: Alert, session: Session) -> Optional[bool]:
+    """
+    Execute alert.sql and return value if any rows are returned
+    """
+    database = alert.database
+    if not database:
+        logger.error("Alert database not preset")
+        return
+
+    if not alert.sql:
+        logger.error("Alert SQL not preset")
+        return
+
+    engine = database.get_sqla_engine()
+    db_engine_spec = database.db_engine_spec
+    parsed_query = ParsedQuery(alert.sql)
+    sql = parsed_query.stripped()
+
+    # update alert.last_eval_dttm
+    alert.last_eval_dttm = datetime.utcnow()
+    session.commit()
+
+    with closing(engine.connect()) as conn:
+        result = conn.execute(sql)
+        return result.rowcount > 0
```

Comment:
potential improvement: if there is a row with one or multiple cells, we could evaluate whether they are falsy and alert if any cell is truthy. This allows for someone to evaluate multiple criteria in a single scan / alert.